src/components/Skeleton/Skeleton.tsx

```diff
@@ -0,0 +1,119 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Skeleton.module.css';
+
+type BaseProps = {
+  /**
+   * CSS class names to augment the layout of the skeleton component. For skeletons,
+   * this is mostly useful to add layout styles (e.g., tailwind CSS margins, flex, etc.)
+   */
+  className?: string;
+  /**
+   * Determine whether there should be an animation in the skeleton state
+   */
+  isAnimating?: boolean;
+};
+
+type SkeletonProps = BaseProps & {
+  width?: number | string;
+  height?: number | string;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Skeleton} from "@chanzuckerberg/eds";`
+ *
+ * Skeleton states allow for hinting at the layout of pending content before the
+ * content is available. The different shapes allow for representation of different
+ * types of content. Mix and match to create a layout that mimicks the pending data.
+ *
+ * Skeletons come in a variety of shapes. The default is a round rect, and there are
+ * others for circular objects, or text layouts. Use any meaningful CSS length values
+ * (e.g., rem, ch, px, etc.)
+ *
+ * Examples:
+ *
+ * For text, Use `Skeleton.Text` and specify the height and the width (for example in 'ch' units).
+ *
+ * For circular objects, use `Skeleton.Circle' and specify the width (which is the same as the height).
+ *
+ * For all other shapes, use `Skeleton.Rect` (or just Skeleton) and specify the width and height.
+ */
+export const Skeleton = ({
+  className,
+  isAnimating = true,
+  height,
+  width,
+  ...other
+}: SkeletonProps) => {
+  const componentClassName = clsx(
+    className,
+    styles['skeleton'],
+    isAnimating && styles['skeleton--is-animating'],
+  );
+  return (
+    <div
+      aria-hidden
```

Comment:
Should we have some sort of message for screen readers saying something like "loading..."?

Along does that at https://github.com/chanzuckerberg/along/blob/3215c989afd8b41fc5960952df1fb4991265844a/client/src/ui/Spinner/Spinner.tsx#L13

Reply:
@ahuth We do have such text for our spinner (`LoadingIndicator`), which can be configured if the default isn't sufficient. For this one, opted to leave it out, as a layout could have many skeleton objects mocking the pre-load UI, and the screen reader would see "loading loading loading loading...". In discussion, we reasoned it's best to document how to add a label on the container, or to use `aria-live` on a page-level component:

- user interacts with some control
- page loads some layout showing these skeleton blocks (instead of a spinner, etc.)
- the interaction should then update the app's `aria-live` text, or could have an `aria-label` on the container that is removed once the data is fully loaded (and the skeletons are replaced with actual content)

I'll add an example of this `aria-label` to storybook to reflect this recommendation